subsys/bluetooth/audio/pacs.c

```diff
@@ -33,145 +33,178 @@
 
 #define PAC_NOTIFY_TIMEOUT	K_MSEC(10)
 
-NET_BUF_SIMPLE_DEFINE_STATIC(read_buf, CONFIG_BT_L2CAP_TX_MTU);
+#define PACS(_name, _work_handler) \
+	struct pacs _name = { \
+		.work = Z_WORK_DELAYABLE_INITIALIZER(_work_handler), \
+	};
+
+#define PACS_LOCATION(_name, _work_handler) \
+	struct pacs_location _name = { \
+		.work = Z_WORK_DELAYABLE_INITIALIZER(_work_handler), \
+	};
 
-static const struct bt_audio_pacs_cb *pacs_cb;
+struct pacs_location {
+	struct k_work_delayable work;
+	uint32_t location;
+};
 
-static void pac_data_add(struct net_buf_simple *buf, uint8_t num,
-			 struct bt_codec_data *data)
+struct pacs {
+	struct k_work_delayable work;
+	sys_slist_t list;
+};
+
+#if defined(CONFIG_BT_PAC_SNK)
+static uint16_t snk_available_contexts;
+static const uint16_t snk_supported_contexts = CONFIG_BT_PACS_SNK_CONTEXT;
+#else
+static const uint16_t snk_available_contexts = BT_AUDIO_CONTEXT_TYPE_PROHIBITED;
+static const uint16_t snk_supported_contexts = BT_AUDIO_CONTEXT_TYPE_PROHIBITED;
+#endif /* CONFIG_BT_PAC_SNK */
+
+#if defined(CONFIG_BT_PAC_SRC)
+static uint16_t src_available_contexts;
+static const uint16_t src_supported_contexts = CONFIG_BT_PACS_SRC_CONTEXT;
+#else
+static const uint16_t src_available_contexts = BT_AUDIO_CONTEXT_TYPE_PROHIBITED;
+static const uint16_t src_supported_contexts = BT_AUDIO_CONTEXT_TYPE_PROHIBITED;
+#endif /* CONFIG_BT_PAC_SRC */
+
+NET_BUF_SIMPLE_DEFINE_STATIC(read_buf, CONFIG_BT_L2CAP_TX_MTU);
+
+static ssize_t pac_data_add(struct net_buf_simple *buf, size_t count,
+			    struct bt_codec_data *data)
 {
-	for (uint8_t i = 0; i < num; i++) {
-		struct bt_pac_codec_capability *cc;
+	size_t len = 0;
+
+	for (size_t i = 0; i < count; i++) {
+		struct bt_pac_ltv *ltv;
 		struct bt_data *d = &data[i].data;
+		const size_t ltv_len = sizeof(*ltv) + d->data_len;
+
+		if (net_buf_simple_tailroom(buf) < ltv_len) {
+			return -ENOMEM;
+		}
 
-		cc = net_buf_simple_add(buf, sizeof(*cc));
-		cc->len = d->data_len + sizeof(cc->type);
-		cc->type = d->type;
+		ltv = net_buf_simple_add(buf, sizeof(*ltv));
+		ltv->len = d->data_len + 1;
```

Comment:
The addition of `1` here, is due to the type, right?

Perhaps
```suggestion
		ltv->len = d->data_len + sizeof(d->type);
```
or
```suggestion
		ltv->len = d->data_len + sizeof(ltv->type);
```
Would make that clearer